server/lib/availabilitySync.ts

```diff
@@ -57,141 +54,93 @@ class AvailabilitySync {
+          // related media is available
+          const requests = await requestRepository
+            .createQueryBuilder('request')
+            .leftJoinAndSelect('request.media', 'media')
+            .where('(media.id = :id)', {
+              id: media.id,
+            })
+            .andWhere(
+              '((request.is4k = 0 AND media.status IN (:...mediaStatus)) OR (request.is4k = 1 AND media.status4k IN (:...mediaStatus)))',
+              {
+                mediaStatus: [
+                  MediaStatus.AVAILABLE,
+                  MediaStatus.PARTIALLY_AVAILABLE,
+                ],
+              }
+            )
+            .getMany();
+
+          // Check if a season is processing or pending to
+          // make sure we set media to the correct status
+          let mediaStatus = MediaStatus.UNKNOWN;
+          let mediaStatus4k = MediaStatus.UNKNOWN;
+
+          if (media.mediaType === 'tv') {
+            mediaStatus = await this.findMediaStatus(requests, false);
+            mediaStatus4k = await this.findMediaStatus(requests, true);
+          }
+
           if (
-            media.status !== MediaStatus.UNKNOWN ||
-            media.status4k !== MediaStatus.UNKNOWN
+            media.status === MediaStatus.AVAILABLE ||
+            media.status === MediaStatus.PARTIALLY_AVAILABLE
           ) {
-            const request = await requestRepository.find({
-              relations: {
-                media: true,
-              },
-              where: { media: { id: media.id } },
-            });
-
-            logger.info(
-              `Media ID ${media.id} does not exist in any of your media instances. Status will be changed to unknown.`,
-              { label: 'AvailabilitySync' }
-            );
-
-            await mediaRepository.update(media.id, {
-              status: MediaStatus.UNKNOWN,
-              status4k: MediaStatus.UNKNOWN,
-              serviceId: null,
-              serviceId4k: null,
-              externalServiceId: null,
-              externalServiceId4k: null,
-              externalServiceSlug: null,
-              externalServiceSlug4k: null,
-              ratingKey: null,
-              ratingKey4k: null,
-            });
-
-            await requestRepository.remove(request);
+            (media.status = mediaStatus),
+              (media.serviceId =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.serviceId
+                  : null),
+              (media.externalServiceId =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceId
+                  : null),
+              (media.externalServiceSlug =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceSlug
+                  : null),
+              (media.ratingKey =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.ratingKey
+                  : null);
```

Comment:
Switched to using query builder to make sure requests are related to only available media. We then check those requests to set the media to the correct status. For example, if the media is deleted and a request still remains, the media should be set to processing.